src/evm-block-extractor/src/block_extractor.rs

```diff
@@ -116,4 +119,35 @@ impl BlockExtractor {
 
         Ok((from_block_inclusive, to_block_inclusive))
     }
+
+    /// Collects blocks from the EVMC and stores them in the database.
+    /// Returns the inclusive range of blocks that were collected.
+    /// This collects also the genesis accounts if needed.
+    async fn collect_genesis_balances(&self) -> anyhow::Result<()> {
+        if self.blockchain.get_genesis_balances().await?.is_some() {
+            log::debug!("Genesis balances already present in the DB. Skipping");
```

Comment:
Prevent unnecessary nesting by returning Ok here and removing else clause

Reply:
fixed